kai/reactive_codeplanner/agent/reflection_agent.py

```diff
@@ -1,73 +1,74 @@
 import json
 import os
 import re
-from dataclasses import dataclass, field
+from dataclasses import dataclass
 from pathlib import Path
 from typing import Any, Optional
 
-import tree_sitter as ts
-import tree_sitter_java as tsj
 from langchain.prompts.chat import (
     AIMessagePromptTemplate,
     HumanMessagePromptTemplate,
     SystemMessagePromptTemplate,
 )
-from langchain_core.messages import AIMessage, HumanMessage, SystemMessage
+from langchain_core.messages import AIMessage, HumanMessage
 
 from kai.llm_interfacing.model_provider import ModelProvider
+from kai.logging.logging import TRACE, get_logger
 from kai.reactive_codeplanner.agent.api import Agent, AgentRequest, AgentResult
 from kai.reactive_codeplanner.agent.ast_diff.parser import Language, extract_ast_info
 
+logger = get_logger(__name__)
+
 
 @dataclass
 class ReflectionTask(AgentRequest):
-    """TODO (pgaikwad): We shouldn't need this class. this is a stop-gap solution as Task is mostly unknown atm"""
-
     # contents of input file prior to invoking previous agent
-    original_file: str = ""
+    original_file_contents: str
     # contents of updated file produced by previous agent
-    updated_file: str = ""
+    updated_file_contents: str
+    # a list of issues originally identified in the file
+    issues: list[str]
     # reasoning produced by previous agent
     reasoning: str = ""
-    # a list of issues originally identified in the file
-    issues: set[str] = field(default_factory=set)
     # a keyword describing target technology to act as a hint to agent
     target_technology: str = "Quarkus"
+    # keyword describing overall application's language
+    app_language: str = "Java"
```

Comment:
is there a way this is calculatable via the file contents?

Reply:
@shawn-hurley yes, we already do that per file, but this is something I imagine coming from the user. This is not for a specific file but the entire application.